src/Chapter3.hs

```diff
@@ -560,22 +691,31 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+
+newtype PlayerHealth = PlayerHealth {phealth ::Int}
+newtype PlayerArmor = PlayerArmor {parmor :: Int}
+newtype PlayerAttack = PlayerAttack {pattack :: Int}
+newtype PlayerDexterity = PlayerDexterity {pdexterity :: Int}
+newtype PlayerStrength = PlayerStrength { pstrength :: Int}
+newtype PlayerDamage = PlayerDamage { pdamage :: Int}
+newtype PlayerDefence = PlayerDefence { pdefence :: Int}
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: PlayerHealth
+    , playerArmor     :: PlayerArmor
+    , playerAttack    :: PlayerAttack
+    , playerDexterity :: PlayerDexterity
+    , playerStrength  :: PlayerStrength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: PlayerAttack -> PlayerStrength -> PlayerDamage
+calculatePlayerDamage attack strength = PlayerDamage ((pattack attack) + (pstrength strength))
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: PlayerArmor -> PlayerDexterity -> PlayerDefence
+calculatePlayerDefense armor dexterity = PlayerDefence ((parmor armor) * (pdexterity dexterity))
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: PlayerDamage -> PlayerDefence -> PlayerHealth -> PlayerHealth
+calculatePlayerHit damage defense health = PlayerHealth ((phealth health) + (pdefence defense) - (pdamage damage))
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype